src/main/java/de/tum/in/www1/artemis/service/connectors/gitlabci/GitLabCIBuildPlanLinkInfoContributor.java

```diff
@@ -0,0 +1,26 @@
+package de.tum.in.www1.artemis.service.connectors.gitlabci;
+
+import java.net.URL;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.actuate.info.Info;
+import org.springframework.boot.actuate.info.InfoContributor;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+import de.tum.in.www1.artemis.config.Constants;
+
+@Component
+@Profile("gitlabci")
+public class GitLabCIBuildPlanLinkInfoContributor implements InfoContributor {
+
+    @Value("${server.url}")
+    private URL artemisServerUrl;
+
+    @Override
+    public void contribute(Info.Builder builder) {
+        // TODO: Not defined in https://github.com/ls1intum/Artemis/blob/develop/src/main/webapp/app/exercises/programming/shared/utils/programming-exercise.utils.ts#L24-L28
```

Comment:
E.g. in the overview page of a programming exercise, there is a link to the template and solution build plan. The URL provided by the server contains wildcards which are replaced in the client.
```typescript
template.replace('{buildPlanId}', buildPlanId).replace('{projectKey}', projectKey)
```
In GitLab CI, there are not `buildPlanId`s nor `projectKey`s, as the build is always mapped to the repository.
With the build plan API, the build plan is accessible over the provided URL, which depends on the id of the programming exercise.
Therefore, the client has to replace the exercise id.
I have not done this yet, because I am only familiar with Java and this is probably a not so important feature.